src/main/kotlin/com/asap/asapbackend/domain/announcement/application/SchoolAnnouncementService.kt

```diff
@@ -0,0 +1,124 @@
+package com.asap.asapbackend.domain.announcement.application
+
+import com.asap.asapbackend.domain.announcement.application.dto.*
+import com.asap.asapbackend.domain.announcement.domain.service.SchoolAnnouncementModifier
+import com.asap.asapbackend.domain.announcement.domain.service.SchoolAnnouncementReader
+import com.asap.asapbackend.domain.child.domain.service.ChildReader
+import com.asap.asapbackend.domain.classroom.domain.model.Classroom
+import com.asap.asapbackend.domain.classroom.domain.service.ClassroomReader
+import com.asap.asapbackend.global.security.getCurrentUserId
+import com.asap.asapbackend.global.security.getTeacherId
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional(readOnly = true)
+class SchoolAnnouncementService(
+    private val schoolAnnouncementReader: SchoolAnnouncementReader,
+    private val schoolAnnouncementModifier: SchoolAnnouncementModifier,
+    private val classroomReader: ClassroomReader,
+    private val childReader: ChildReader,
+) {
+
+    fun getSchoolAnnouncementSlice(request: GetSchoolAnnouncementSlice.Request): GetSchoolAnnouncementSlice.Response {
+        val classroom = getChildClassroom()
+        val schoolAnnouncements = schoolAnnouncementReader.findAllSchoolAnnouncements(
+            classroom.id,
+            PageRequest.of(request.page, request.size)
+        )
+        return GetSchoolAnnouncementSlice.convertSchoolAnnouncementToResponse(schoolAnnouncements) {
+            schoolAnnouncementReader.findSchoolAnnouncementCategory(it, classroom.id)
```

Comment:
```suggestion
            schoolAnnouncementReader.findSchoolAnnouncementCategory(schoolAnnouncementId = it, classId = classroom.id)
```
이게 by가 없으니까 이런식으로 라도 써주는거 어때

Reply:
오케~

Reply:
명시하기에는 다 추가해줘야해서 메소드 이름을 바꿀게

Reply:
또 그렇기에는 메소드 이름이 너무 길어져서 그냥 이렇게 놔두는거는 어때?